### PR TITLE
CA-585 Clicked on the grid view icon on the top right and I see myself twice.

### PR DESCRIPTION
**Pull request by giladadon1997**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### JIRA link(s):

- [AHOYAPPS-0000](https://issues.corp.twilio.com/browse/AHOYAPPS-0000)

### Description

A description of what this PR does.

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with `npm test`
* [ ] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [ ] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Re-tested if necessary

### CODE REVIEW

File: src/components/MenuBar/ToggleGridViewButton/ToggleGridViewButton.tsx [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
+  };
 
-    return (
-        <IconButton onClick={() => setGridView(!gridView)}>
-            {gridView ? <GridOffIcon /> : <GridOnIcon />}
-        </IconButton>
-    );
+  return (
+    <IconButton
+      onClick={() => {
```
Comment (Galzk) [inline src/components/MenuBar/ToggleGridViewButton/ToggleGridViewButton.tsx]: change the onClick={...} to:
onClick={handleChange}

**Review by Galzk — CHANGES_REQUESTED** [review]
Nice work!
I'd also add that if the viewPort is smaller than x, then the toggle will be between collaboration and grid (which is only 2 options)
Because on smaller screens the button doesn't work well
Consider adding ViewModeArrayMobile and add a condition to when to use what

File: src/components/MenuBar/ToggleGridViewButton/ToggleGridViewButton.tsx (hunk 1 above)
Comment (giladadon1997) [reply src/components/MenuBar/ToggleGridViewButton/ToggleGridViewButton.tsx]: done

File: src/state/settings/settingsReducer.ts [hunk 2]
```diff
@@ -1,56 +1,63 @@
...
+export interface SettingsAction {
+  name: SettingsKeys;
+  value: string;
+}
+
+export const initialSettings: Settings = {
+  trackSwitchOffMode: undefined,
+  dominantSpeakerPriority: 'standard',
+  viewMode: VIEW_MODE.default,
+  maxTracks: isMobile ? '5' : '10',
```
Comment (talr) [inline src/state/settings/settingsReducer.ts:40]: Shall we change it to 4 and and 9?

File: src/state/settings/settingsReducer.ts [hunk 3]
```diff
@@ -1,56 +1,63 @@
...
+  renderDimensionLow?: RenderDimensionValue;
+  renderDimensionStandard?: RenderDimensionValue;
+  renderDimensionHigh?: RenderDimensionValue;
+}
+
+type SettingsKeys = keyof Settings;
+export const VIEW_MODE = {
+  grid_2_2: 'grid 2X2',
+  grid_3_3: 'grid 3X3',
+  grid_4_4: 'grid 4X4',
```
Comment (talr) [inline src/state/settings/settingsReducer.ts]: columns_2
columns_3
columns_4

File: src/components/ParticipantStrip/ParticipantStripGrid/ParticipantStripGrid.tsx [hunk 4]
```diff
@@ -50,6 +50,9 @@ export default function ParticipantStripGrid({ viewMode }: ParticipantStripGridP
       } else if (viewMode.includes('4X4')) {
         setMdState(3);
         setLgState(3);
+      } else {
+        setMdState(3);
+        setLgState(4);
```
Comment (talr) [inline src/components/ParticipantStrip/ParticipantStripGrid/ParticipantStripGrid.tsx]: Opposite

Comment (Galzk) [inline src/components/ParticipantStrip/ParticipantStripGrid/ParticipantStripGrid.tsx]:
```suggestion
        setMdState(4);
        setLgState(3);

```

**Review by Galzk — CHANGES_REQUESTED** [review]

File: src/components/MenuBar/ConnectionOptions/ConnectionOptions.tsx [hunk 5]
```diff
@@ -74,6 +74,7 @@ export default function ConnectionOptions({ className, hidden }: { className?: s
               <MenuItem value="grid 3X3">3 Column</MenuItem>
               <MenuItem value="grid 4X4">4 Column</MenuItem>
```
Comment (talr) [inline src/components/MenuBar/ConnectionOptions/ConnectionOptions.tsx]: 3 Columns
4 Columns

File: src/components/MenuBar/ToggleGridViewButton/ToggleGridViewButton.test.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  it('should render correctly', () => {
+    const wrapper = shallow(<ToggleGridViewButton />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should render the next view mode', () => {
+    const wrapper = shallow(<ToggleGridViewButton />);
+    wrapper.find(IconButton).simulate('click');
+    expect(mockDispatchSetting).toHaveBeenCalledWith({ name: 'viewMode', value: 'grid 2X2' });
```
Comment (talr) [inline src/components/MenuBar/ToggleGridViewButton/ToggleGridViewButton.test.tsx]: Same

**Review by talr — CHANGES_REQUESTED** [review]
Tooltip

**Pull request merged**